apps/backoffice-v2/src/common/components/molecules/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,309 @@
+import React, {
+  ComponentProps,
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  Cell,
+  ColumnDef,
+  ExpandedState,
+  flexRender,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getSortedRowModel,
+  OnChangeFn,
+  RowData,
+  RowSelectionState,
+  SortingState,
+  TableOptions,
+  useReactTable,
+} from '@tanstack/react-table';
+import { ScrollArea } from '@/common/components/molecules/ScrollArea/ScrollArea';
+import { DefaultCell } from '@/lib/blocks/components/TableCell/DefaultCell';
+import {
+  Table,
+  TableBody,
+  TableCaption,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from '@/common/components/atoms/Table';
+import { ctw } from '@/common/utils/ctw/ctw';
+import { CollapsibleContent } from '@/common/components/molecules/Collapsible/Collapsible.Content';
+import { Collapsible } from '@/common/components/molecules/Collapsible/Collapsible';
+import { useSort } from '@/common/hooks/useSort/useSort';
+import { isInstanceOfFunction } from '@/common/utils/is-instance-of-function/is-instance-of-function';
+import { checkIsBooleanishRecord } from '@/lib/zod/utils/checkers';
+import { useSelect } from '@/common/hooks/useSelect/useSelect';
+import { ChevronDown } from 'lucide-react';
+import { FunctionComponentWithChildren } from '@/common/types';
+
+export interface IDataTableProps<TData, TValue = any> {
+  data: TData[];
+  sortByField?: string;
+  enableSorting?: boolean;
+  columns: Array<ColumnDef<TData, TValue>>;
+  caption?: ComponentProps<typeof TableCaption>['children'];
+
+  CellWrapper?: FunctionComponentWithChildren<{ cell: Cell<TData, TValue> }>;
+  CollapsibleComponent?: FunctionComponent<{ row: TData }>;
+
+  // Component props
+  props?: {
+    scroll?: Partial<ComponentProps<typeof ScrollArea>>;
+    table?: ComponentProps<typeof Table>;
+    header?: ComponentProps<typeof TableHeader>;
+    head?: ComponentProps<typeof TableHead>;
+    row?: ComponentProps<typeof TableRow>;
+    body?: ComponentProps<typeof TableBody>;
+    cell?: ComponentProps<typeof TableCell>;
+    caption?: ComponentProps<typeof TableCaption>;
+  };
+
+  // react-table options
+  options?: Omit<TableOptions<TData>, 'getCoreRowModel' | 'data' | 'columns'>;
+}
+
+export const DataTable = <TData extends RowData, TValue = unknown>({
```

Comment:
`TValue` is already `any` in another place. If `any` is the one that works let's do it in here as well.